src/core/mac/mac.cpp

```diff
@@ -2096,9 +2325,10 @@ otError Mac::AppendHeaderIe(bool aIsTimeSync, TxFrame &aFrame)
 }
 #endif // OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
 
-void Mac::UpdateFrameControlField(bool aIsTimeSync, uint16_t &aFcf)
+void Mac::UpdateFrameControlField(Neighbor *aNeighbor, bool aIsTimeSync, uint16_t &aFcf) const
```

Comment:
Can we declare `aNeighbor` as const?
```suggestion
void Mac::UpdateFrameControlField(const Neighbor *aNeighbor, bool aIsTimeSync, uint16_t &aFcf) const
```